### PR TITLE
Fix issue 224 - pspm_data_editor

### LINKED ISSUE #224: Data editor to show missing epochs in time order
### Feature Description
Data editor is currently showing missing epochs based on the default order. Users may prefer to see the missing epochs listed in the time order.

### PR DESCRIPTION
**Pull request by teddychao**
Fixes #224 .

Changes proposed in this pull request:
- The epoch list in `pspm_data_editor` now sorts the epoch list based on the time order.
- The plots in `pspm_data_editor` now display the units for the x and y axises.

### CODE REVIEW

File: src/pspm_data_editor.m [hunk 1]
```diff
@@ -838,7 +841,13 @@ function UpdateEpochList(hObject)
...
   end
   set(handles.lbEpochs, 'String', names);
-  handles.epochs = epochs;
+  % update to sort epochs according to the time series order
+  epochs_list = 1:length(epochs);
+  for i_epochs_list = 1:length(epochs) % obtain data
+    epochs_list(i_epochs_list) = epochs{1,i_epochs_list}.range(1);
+  end
+  [~,idx]=sort(epochs_list);
+  handles.epochs = epochs{1,idx};
```
Comment (jbrochar) [inline src/pspm_data_editor.m:850]: FYI I think the six previous lines can be shorten as:
`handles.epochs = sort( cellfun(@(epoch) epoch.range(1), epochs) )`
Which might be less intuitive to read or not. I let you be the judge of that

**Review by jbrochar — COMMENTED** [review]
Good from a coding perspective.
I added a minor suggestion.

File: src/pspm_data_editor.m (hunk 1 above)
Comment (teddychao) [reply src/pspm_data_editor.m:850]: Great idea, thanks Jules!

**Review by jbrochar — APPROVED** [review]

File: src/pspm_data_editor.m (hunk 1 above)
Comment (jbrochar) [reply src/pspm_data_editor.m:850]: Oh? There might be a typo from my end then.
What is the output of `cellfun(@(epoch) epoch.range(1), epochs)`?
Alternatively, can you send me be email (jules.brochard@gmail.com) an example of the variable `epochs`?

Comment (teddychao) [reply src/pspm_data_editor.m:850]: > Oh? There might be a typo from my end then.
> What is the output of `cellfun(@(epoch) epoch.range(1), epochs)`?
> Alternatively, can you send me be email ([jules.brochard@gmail.com](mailto:jules.brochard@gmail.com)) an example of the variable `epochs`?

No problem, here are two screenshots showing the results... I will send you the link to the test files soon.
original solution
Screenshot 2022-06-03 at 16 48 17
proposed solution
Screenshot 2022-06-03 at 16 47 28

Comment (jbrochar) [reply src/pspm_data_editor.m:850]: it is strange indeed.

What is the output of `cellfun(@(epoch) epoch.range(1), epochs)`? 
This should simply extract the .range(1) field of every element of the cell array `epochs`.

Could you also try `cellfun(@(epoch) epoch.range(1), epochs(1,:))` ?

**Pull request merged**